src/main/java/aztech/modern_industrialization/machines/multiblocks/ShapeMatcher.java

```diff
@@ -47,14 +47,14 @@ public ShapeMatcher(Level world, BlockPos controllerPos, Direction controllerDir
         this.hatchFlags = toWorldPos(controllerPos, controllerDirection, template.hatchFlags);
     }
 
```

Comment:
Why make all of this stuff `protected`?

Reply:
I use them in my [custom shape matcher](https://github.com/Swedz/Extended-Industrialization/blob/tesla-tower/src/main/java/net/swedz/extended_industrialization/machines/blockentity/multiblock/teslatower/SameCableTierShapeMatcher.java) to prevent the tesla tower from linking when there is more than one tier of energy hatch present

Reply:
The main reason I override the rematch method is because it prevents needlessly having to iterate over the same stuff twice but I suppose it's such a small dataset it doesn't matter too much. I can make that change

Reply:
Maybe the better option is moving the simpleMembers loop to a checkRematch method? That way the members don't need to be iterated over more than once and still avoids needing to override rematch.

Reply:
What makes you think that iterating over the members a second time is a potential performance issue?

Reply:
Depends on the number of members but I know some people have made some rather large multiblocks. Requesting world info about block states and block entities more than once for the same check is kind of unnecessary, no?

Reply:
Depends what you do with it. If you're just checking for hatch tiers, it should be enough to iterate over `matchedHatches`.